documentation/rest-clients.md

```diff
@@ -0,0 +1,158 @@
+
+Both methods fully harness the power of `@azure/core-rest-pipeline` and `@azure/identity`.
+
+The basic structure of calls with REST clients is:
+
+1. Initialize your client
+2. Send the request
+3. Handle the response
+
+We will go into each step in the following sections
+
+## 1. Initialize the Client
+First import the client
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+```
+
+Most clients require authenticating through their `credential` parameter. Depending on what authentication support your library is using, you can either authenticate with AAD or authenticate with an AzureKeyCredential.
+
+Additionally, most of our clients accept an `endpoint` parameter at initialization, usually a link to your own resource.
+
+### Authenticating with AAD
+Authenticating with AAD
+Depending on your library, our clients support authenticating with an Azure Active Directory (AAD) token credential. We always recommend using a credential type obtained from the `@azure/identity` library for AAD authentication. For this example, we use the most common DefaultAzureCredential.
+
+As an installation note, the `@azure/identity` library is not a dependency of this library. Please run `npm install @azure/identity` before using AAD authentication
+
+The following code snippet shows you how to authenticate with a DefaultAzureCredential.
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient("https://example.org/", new DefaultAzureCredential());
+```
+
+## 2. Send a request
+
+Once the client has been initialized we need to set a path to work with. For this, the REST client exposes 2 functions `path` and `pathUnchecked`
+
+### Path
+Path function takes a path as the first parameter and accepts any path defined in the Swagger. It also detects if the path needs a path parameter and requires them as positional parameters to `path`. Once the path is set users can access functions for all the supported verbs on that path
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient ("https://example.org/", new DefaultAzureCredential());
+const response = await client.path("/hello/{name}", "Brian").get();
+
+// response.body is strongly typed
+console.log(response.body);
+// {content: "Hello Brian"}
+```
+
+### PathUnchecked
+PathUnchecked function is similar to Path, it takes a path as the first parameter, this can be any arbitrary path. It also detects if the path needs a path parameter and requires them as positional parameters to `pathUnchecked`. Once the path is set users can access functions for any verb on that path.
+
+The main difference with `path` is that `pathUnchecked` doesn't have strongly typed payload, headers, or query parameters and has `any` as the response type.
```

Comment:
I know more and more of the world is TypeScript, but are the interesting differences between path and pathUnchecked in JavaScript?  I think when you are using raw JS these two are basically the same because the differences are just around typing of parameters and the return value?

Reply:
It depends on the dev environment they use. For example, if you are writing an app using plain JavaScript in VSCode, VSCode will load the library's types and provide IntelliSense for you. So most of the benefits still apply to users writing code in editors that use TS Types for IntelliSense.

But I think you bring up a good point! We may want to mention that JS users will still get benefits from this and not just limited to TS.

Reply:
Love it!